test/functional/helpers/ci-metrics.js

```diff
@@ -0,0 +1,105 @@
+import wd from 'wd';
+import { logger, fs, mkdirp } from 'appium-support';
+import request from 'request-promise';
+import { exec } from 'teen_process';
+import _ from 'lodash';
+import path from 'path';
+import uuid from 'uuid';
+import { asyncify } from 'asyncbox';
+
+
+/*
+ TODO: Get Git-sha for this commit, top add to metrics being sent so that
+       the sense can be made at a finer grained level
+ */
+
+
+const log = logger.getLogger('CI METRICS');
+
+async function sendToSumoLogic (events) {
+  let opts = {
+    method: 'POST',
+    uri: process.env.SUMO_LOGIC_ENDPOINT, // defined in .travis.yml
+    body: events,
+    json: true,
+  };
+
+  return await request(opts);
+}
+
+const getGitRev = _.memoize (
+  async function gitDetails () {
+    let {stdout} = await exec('git', ['rev-parse', 'HEAD']);
+    return (stdout || '').trim();
+  }
+);
+
+if (process.env.CI_METRICS) {
```

Comment:
it looks like this block will run every time we `require` this file? i know that internals of node might mean this only happens once, but i wonder if it would be cleaner to pull this into a `patchDriverWithEvents` method or similar, which is exported and called rather than simply running on require?